mpc-core/src/protocols/rep3.rs

```diff
@@ -378,6 +427,14 @@ impl<F: PrimeField, N: Rep3Network> PrimeFieldMpcProtocol<F> for Rep3Protocol<F,
         Ok(a.a + a.b + c)
     }
 
+    fn open_many(&mut self, a: &[Self::FieldShare]) -> std::io::Result<Vec<F>> {
+        let bs = a.iter().map(|x| x.b).collect_vec();
+        self.network.send_next(bs)?;
+        let cs = self.network.recv_prev::<Vec<F>>()?;
+
+        Ok(izip!(a, cs).map(|(x, c)| x.a + x.b + c).collect_vec())
```

Comment:
re-use `cs` as the result buffer to not have to allocate

Reply:
rewritten